drivers/shared/executor/server.go

```diff
@@ -87,20 +87,46 @@ func (s *grpcExecutorServer) Version(context.Context, *proto.VersionRequest) (*p
 	}, nil
 }
 
-func (s *grpcExecutorServer) Stats(context.Context, *proto.StatsRequest) (*proto.StatsResponse, error) {
-	stats, err := s.impl.Stats()
-	if err != nil {
-		return nil, err
+func (s *grpcExecutorServer) Stats(req *proto.StatsRequest, stream proto.Executor_StatsServer) error {
+	ctx := stream.Context()
+
+	interval := time.Duration(req.Interval)
+	if interval.Nanoseconds() == 0 {
```

Comment:
The .Nanoseconds() call is unnecessary when comparing against a const, but I'm sure the compiler emits the same code either way.